added editor security to vault pages (#69335)
ansible/ansible: docs/docsite/rst/user_guide/vault.rst

For best practices advice, refer to :ref:`best_practices_for_variables_and_vaults`.

.. note::

>     By default vault uses your ``$EDITOR`` when creating, editing and viewing a file, be aware that certain editors will keep copies of the data either in swap, backup or clipboard files.

Reviewer: ```suggestion
    By default, vault uses your ``$EDITOR`` when creating, editing and viewing a file. Be aware that certain editors will keep copies of the data either in swap, backup or clipboard files.
```

>     It is up to the user to secure their editor while using vaults, we have added some common sense settings below.

Reviewer: ```suggestion
    While you are responsible for  securing your editor while using vaults, we have added some common sense settings below.
```

only one vault-id can be used.

.. _vault_securing_editor:

> Securing your editor

Reviewer: ```suggestion

Securing your editor
```

````````````````````

> As noted above, the vault system relies on your configured ``$EDITOR`` which can be a source of disclosures, most editors have conviniences to prevent loss of data, but this normaly relies

Reviewer: ```suggestion
As noted above, the vault system relies on your configured ``$EDITOR`` which can be a source of disclosures. Most editors have ways to prevent loss of data, but these normally rely
```

on extra plain text files that can have a clear text copy of your secrets. To prevent this you should consult your editor's manual to ensure you configure it to avoid disclosing secure data.

vim
^^^

> These are some well known options you can set in command mode to avoid cases of disclosure, be aware that there are probably more settings you'll need to modify to ensure security, specially when using plugins.

Reviewer: ```suggestion
These are some well known options you can set in command mode to avoid cases of disclosure. Be aware that there are probably more settings you'll need to modify to ensure security, especially when using plugins.
```

> The swapfiles are a kind of autosave in case of crash/interruption, you can disable this via::

Reviewer: ```suggestion
The swapfiles are a kind of autosave in case of crash/interruption. Disable them as follows::
```

    set noswapfile

Prevent creating backup files::

    set nobackup
    set nowritebackup

> The viminfo file copies data from your session, sometimes even secrets, but you can disable this for the current session like this::

Reviewer: ```suggestion
The viminfo file copies data from your session, sometimes even secrets, but you can disable this for the current session as follows::
```

    set viminfo=

Disable copying to the system clipboard::

    set clipboard=

> Most of these you can also set in ``.vimrc`` for all files or just specific paths/extensions, see the vim manual for specifics.

Reviewer: ```suggestion
Most of these you can also set in ``.vimrc`` for all files or just specific paths/extensions. See the vim manual for specifics.
```

> emacs

Reviewer: ```suggestion
Emacs
```

^^^^^

> These options can be used to avoid common known cases of disclosure, this list is by now way complete, specially when you consider plugins and extensions.

Reviewer: ```suggestion
Use these options to avoid common known cases of disclosure,  but be aware that this list is note complete, especially when you consider plugins and extensions. 
```
